src/main/java/frc/robot/subsystems/RollerSubsystem.java

```diff
@@ -146,29 +151,48 @@ public void periodic() {
         }
 
         // Otherwise, open if we're opening and close if we're closing.
-        if (openTimer.hasStarted()) openMotor.set(0.5);
-        else if (closeTimer.hasStarted()) openMotor.set(-0.2);
-        else openMotor.set(0);
+        if (openTimer.hasStarted()) openRoller();
+        else if (closeTimer.hasStarted()) closeRoller();
+        else if (heldPiece == HeldPiece.CONE) activeclampRoller();
+        else neitherRoller();;
+    }
 
-        // if wheels must intake, and the limit switch is not pressed, turn on motors
-        if (limitSwitch.get()) {
-            leftBeak.set(rollPower);
-        } else {
-            leftBeak.set(Math.min(rollPower, 0.0));
-        }
+    protected void openRoller(){
+        openMotor.set(0.5);
+    }
+
+    protected void activeclampRoller(){
+        openMotor.setVoltage(-2);
+    }
 
+    protected void closeRoller(){
+        openMotor.set(-0.2);
+    }
+
+    protected void neitherRoller(){
+        openMotor.set(0);
+    }
```

Comment:
these methods could probably be renamed to be more clear. for example i'm not sure if 'roller' should even be in the name.. unless roller means the intake as a whole?